From 604bd93b3f0ca12c8113ed1d5001e7845fe9fe8c Mon Sep 17 00:00:00 2001
From: Franck Nijhof <opensource@frenck.dev>
Date: Fri, 21 Aug 2026 15:05:00 +0200
Subject: [PATCH 2/4] Patch redirect logs to docker output

Sends the NGINX access and error logs to the container's stdout so they
show up in the add-on log, instead of writing them to files the user
cannot easily reach.

Covers the split of conf.d/include/log.conf into log-proxy.conf and
log-stream.conf introduced upstream in v2.13.0.
---
 backend/templates/dead_host.conf                       | 4 ++--
 backend/templates/default.conf                         | 4 ++--
 backend/templates/letsencrypt-request.conf             | 4 ++--
 backend/templates/proxy_host.conf                      | 4 ++--
 backend/templates/redirection_host.conf                | 4 ++--
 backend/templates/stream.conf                          | 8 ++++----
 docker/rootfs/etc/nginx/conf.d/default.conf            | 6 +++---
 docker/rootfs/etc/nginx/conf.d/include/log-proxy.conf  | 2 +-
 docker/rootfs/etc/nginx/conf.d/include/log-stream.conf | 2 +-
 docker/rootfs/etc/nginx/nginx.conf                     | 2 +-
 10 files changed, 20 insertions(+), 20 deletions(-)

diff --git a/backend/templates/dead_host.conf b/backend/templates/dead_host.conf
index b7f5209..ec881dc 100644
--- a/backend/templates/dead_host.conf
+++ b/backend/templates/dead_host.conf
@@ -10,8 +10,8 @@ server {
 {% include "_hsts.conf" %}
 {% include "_forced_ssl.conf" %}
 
-  access_log /config/logs/dead-host-{{ id }}_access.log standard;
-  error_log /config/logs/dead-host-{{ id }}_error.log warn;
+  access_log /proc/1/fd/1 standard;
+  error_log /proc/1/fd/1 warn;
 
 {{ advanced_config }}
 
diff --git a/backend/templates/default.conf b/backend/templates/default.conf
index 96174e0..cb69592 100644
--- a/backend/templates/default.conf
+++ b/backend/templates/default.conf
@@ -12,8 +12,8 @@ server {
   #listen [::]:80 default;
 {% endif %}
   server_name default-host.localhost;
-  access_log /config/logs/default-host_access.log combined;
-  error_log /config/logs/default-host_error.log warn;
+  access_log /proc/1/fd/1 combined;
+  error_log /proc/1/fd/1 warn;
 {% include "_exploits.conf" %}
 
   include conf.d/include/letsencrypt-acme-challenge.conf;
diff --git a/backend/templates/letsencrypt-request.conf b/backend/templates/letsencrypt-request.conf
index 6bb985b..675b874 100644
--- a/backend/templates/letsencrypt-request.conf
+++ b/backend/templates/letsencrypt-request.conf
@@ -8,8 +8,8 @@ server {
 
   server_name {{ domain_names | join: " " }};
 
-  access_log /config/logs/letsencrypt-requests_access.log standard;
-  error_log /config/logs/letsencrypt-requests_error.log warn;
+  access_log /proc/1/fd/1 standard;
+  error_log /proc/1/fd/1 warn;
 
   include conf.d/include/letsencrypt-acme-challenge.conf;
 
diff --git a/backend/templates/proxy_host.conf b/backend/templates/proxy_host.conf
index 0de6647..c3c7490 100644
--- a/backend/templates/proxy_host.conf
+++ b/backend/templates/proxy_host.conf
@@ -22,8 +22,8 @@ proxy_set_header Connection $http_connection;
 proxy_http_version 1.1;
 {% endif %}
 
-  access_log /config/logs/proxy-host-{{ id }}_access.log proxy;
-  error_log /config/logs/proxy-host-{{ id }}_error.log warn;
+  access_log /proc/1/fd/1 proxy;
+  error_log /proc/1/fd/1 warn;
 
 {{ advanced_config }}
 
diff --git a/backend/templates/redirection_host.conf b/backend/templates/redirection_host.conf
index 61ee0c2..3377364 100644
--- a/backend/templates/redirection_host.conf
+++ b/backend/templates/redirection_host.conf
@@ -12,8 +12,8 @@ server {
 {% include "_hsts.conf" %}
 {% include "_forced_ssl.conf" %}
 
-  access_log /config/logs/redirection-host-{{ id }}_access.log standard;
-  error_log /config/logs/redirection-host-{{ id }}_error.log warn;
+  access_log /proc/1/fd/1 standard;
+  error_log /proc/1/fd/1 warn;
 
 {{ advanced_config }}
 
diff --git a/backend/templates/stream.conf b/backend/templates/stream.conf
index cd15aa4..4a421ba 100644
--- a/backend/templates/stream.conf
+++ b/backend/templates/stream.conf
@@ -12,8 +12,8 @@ server {
 
   proxy_pass {{ forwarding_host }}:{{ forwarding_port }};
 
-  access_log /config/logs/stream-{{ id }}_access.log stream;
-  error_log /config/logs/stream-{{ id }}_error.log warn;
+  access_log /proc/1/fd/1 stream;
+  error_log /proc/1/fd/1 warn;
 
   # Custom
   include /config/nginx/custom/server_stream[.]conf;
@@ -28,8 +28,8 @@ server {
 
   proxy_pass {{ forwarding_host }}:{{ forwarding_port }};
 
-  access_log /config/logs/stream-{{ id }}_access.log stream;
-  error_log /config/logs/stream-{{ id }}_error.log warn;
+  access_log /proc/1/fd/1 stream;
+  error_log /proc/1/fd/1 warn;
 
   # Custom
   include /config/nginx/custom/server_stream[.]conf;
diff --git a/docker/rootfs/etc/nginx/conf.d/default.conf b/docker/rootfs/etc/nginx/conf.d/default.conf
index 2637dfa..4a6108a 100644
--- a/docker/rootfs/etc/nginx/conf.d/default.conf
+++ b/docker/rootfs/etc/nginx/conf.d/default.conf
@@ -8,8 +8,8 @@ server {
 	set $port "80";
 
 	server_name localhost-nginx-proxy-manager;
-	access_log /config/logs/fallback_http_access.log standard;
-	error_log /config/logs/fallback_http_error.log warn;
+	access_log /proc/1/fd/1 standard;
+	error_log /proc/1/fd/1 warn;
 	include conf.d/include/assets.conf;
 	include conf.d/include/block-exploits.conf;
 	include conf.d/include/letsencrypt-acme-challenge.conf;
@@ -30,7 +30,7 @@ server {
 	set $port "443";
 
 	server_name localhost;
-	access_log /config/logs/fallback_http_access.log standard;
+	access_log /proc/1/fd/1 standard;
 	error_log /dev/null crit;
 	include conf.d/include/ssl-ciphers.conf;
 	ssl_reject_handshake on;
diff --git a/docker/rootfs/etc/nginx/conf.d/include/log-proxy.conf b/docker/rootfs/etc/nginx/conf.d/include/log-proxy.conf
index 00ff31c..1877713 100644
--- a/docker/rootfs/etc/nginx/conf.d/include/log-proxy.conf
+++ b/docker/rootfs/etc/nginx/conf.d/include/log-proxy.conf
@@ -1,4 +1,4 @@
 log_format proxy '[$time_local] $upstream_cache_status $upstream_status $status - $request_method $scheme $host "$request_uri" [Client $remote_addr] [Length $body_bytes_sent] [Gzip $gzip_ratio] [Sent-to $server] "$http_user_agent" "$http_referer"';
 log_format standard '[$time_local] $status - $request_method $scheme $host "$request_uri" [Client $remote_addr] [Length $body_bytes_sent] [Gzip $gzip_ratio] "$http_user_agent" "$http_referer"';
 
-access_log /config/logs/fallback_http_access.log proxy;
+access_log /proc/1/fd/1 proxy;
diff --git a/docker/rootfs/etc/nginx/conf.d/include/log-stream.conf b/docker/rootfs/etc/nginx/conf.d/include/log-stream.conf
index 0c1946d..c3581e5 100644
--- a/docker/rootfs/etc/nginx/conf.d/include/log-stream.conf
+++ b/docker/rootfs/etc/nginx/conf.d/include/log-stream.conf
@@ -1,3 +1,3 @@
 log_format stream '[$time_local] [Client $remote_addr:$remote_port] $protocol $status $bytes_sent $bytes_received $session_time [Sent-to $upstream_addr] [Sent $upstream_bytes_sent] [Received $upstream_bytes_received] [Time $upstream_connect_time] $ssl_protocol $ssl_cipher';
 
-access_log /config/logs/fallback_stream_access.log stream;
+access_log /proc/1/fd/1 stream;
diff --git a/docker/rootfs/etc/nginx/nginx.conf b/docker/rootfs/etc/nginx/nginx.conf
index 83cffa7..ba39d9b 100644
--- a/docker/rootfs/etc/nginx/nginx.conf
+++ b/docker/rootfs/etc/nginx/nginx.conf
@@ -9,7 +9,7 @@ worker_processes auto;
 # Enables the use of JIT for regular expressions to speed-up their processing.
 pcre_jit on;
 
-error_log /config/logs/fallback_error.log warn;
+error_log /proc/1/fd/1 warn;
 
 # Includes files with directives to load dynamic modules.
 include /etc/nginx/modules/*.conf;
-- 
2.39.5

